### ISSUE TITLE
SAB and eFax

### ISSUE THREAD

**Issue by kdjstudios** [kdjstudios/SABillingGitlab#260]
In GitLab by @kdjstudios on Mar 3, 2017, 11:06

HD setup efax accounts: http://www.servicedesk.answernet.com/profiles/ticket/2019-04-15-67819

**`Need to check all fax number has a valid on the account while releasing.(Only for Avaaz team)`**

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Mar 6, 2017, 08:34

@smasih @abakshi @tim.traylor 

We are currently switching over to an eFax system through Vitality and we need to have this implemented into SAB. We will most likely need to discuss this prior to the weekly meeting.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jan 29, 2019, 13:37

@amishra @smasih @pchaudhary @tim.traylor 

Marking this for review with @tim.traylor.

Items to review:

- Do we want site to be able to specify which efax to use? a site setting or an admin setting?
- I would believe we use email to fax already. Other wise I am sure client would not be receiving their fax invoices.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jan 31, 2019, 08:53

Reviewed with Tim, we will eventually want to transition over, but will require further discussion. This is not prioritized yet.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Feb 1, 2019, 12:21

@tim.traylor realized today that the Faxing from SA Billing is probably not working. I tested and was able to confirm that I did not received either of two test faxes I sent from the system.

@smasih @amishra @pchaudhary 

We need to have SA Billing Faxing updated/fixed ASAP. This is also related to #311 which has further information on it how we will want this to function. marking this as primary. 

We will need to do the following:
- Find when faxing quick working; also find out why we were not notified of failed faxes. 
- Find all effected accounts and notify the sites/clients accordingly.
- Implement the new faxing system (Further discussion and details with mocks coming soon)
- Test and release.

**Comment by kdjstudios** [comment]
In GitLab by @tim.traylor on Feb 14, 2019, 06:29

@smasih The first item is to determine of any faxing is setup on any accounts at all.

**Comment by kdjstudios** [comment]
In GitLab by @pchaudhary on Mar 27, 2019, 01:50

assigned to @amishra

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Apr 2, 2019, 12:03

This was reviewed today. @tim.traylor is looking further into this, but we believe we should be able to use a standard eFax solution to resolve this. I believe the main question is as what is pointed on in ticket #311 does each site need to use a personalized fax number, or can all sites/clients use the same fax number?

**Comment by kdjstudios** [comment]
In GitLab by @tim.traylor on Apr 9, 2019, 10:35

@smasih Sumeet,  please give us an estimate on converting the current invoice.rb code that sends a fax from the command line execution to simply sending an email.  We'll need:
1. Add outbound_efax_host to the leagal_entity collection.  This string value will contain the hostname portion of the email the 'fax' is sent to.  (This should be added to the UI as well as the collection.).
1. Instead of cmd line executing sendfax it should just send an email to "RecipientFaxNumber@outbound_efax_host" with a PDF attached.  The "@outbound_efax_host" portion being the value looked up for the legal entity.
1. The remainder of the logging for sending a fax can remain in place.

**Comment by kdjstudios** [comment]
In GitLab by @amishra on Apr 11, 2019, 05:12

@kdjstudios 

Thank you for the update.

Estimate:

|Size|Lower Time Bound|Variability|Upper Time Bound|Testing|Time for Testing| Time for Dev & Testing|
|-|-|-|-|-|-|-|
|R-Small|2|Aware of Concepts, implementation tricky|3.5|Low Impact|1.0|4.5|

Thanks.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Apr 11, 2019, 11:55

@amishra 

I am going to approve this, as we will need to have the faxing working.

**Comment by kdjstudios** [comment]
In GitLab by @amishra on Apr 12, 2019, 06:12

@kdjstudios 

Thank you for the update.

For sending an email for fax, we are using the same template as the send email for invoices.
Please see attachment.
![image](/uploads/944324b423e3d2cc54a5f86abb5c86f6/image.png)


If you need to change the subject/body of the email or something else then please let us know an also provide me the verbiage.

Thanks

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Apr 12, 2019, 09:48

@amishra @tim.traylor 

Thank you I believe this same format will work as I am sure the body is ignored and the subject line is just added to the header of the fax; while the attached document is what is actually getting faxed; but I will let @tim.traylor confirm.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Apr 16, 2019, 10:27

I will put in a HD ticket to have the eFax setup for internal and external. i will follow up with the info once it is created.  

Where are we at with development on this? If needed please use "FAXNUMBER@billings.fax.answernet.com" to test sending faxes out. My office fax is 406-534-0399 in case you need a fax number to test.

**Comment by kdjstudios** [comment]
In GitLab by @amishra on Apr 17, 2019, 08:41

@kdjstudios 

Thank you for the update.



> Where are we at with development on this?

We have completed the development and some testing is pending. We will let you know when ready for your testing.

Thanks.

**Comment by kdjstudios** [comment]
In GitLab by @amishra on Apr 17, 2019, 23:15

@kdjstudios 

We have set up on the dev server for your testing. Please test it.

**Step to test**

* Goto Admin section and select Legal Entities.
* Create a new Legal Entities or edit Legal Entities as you want.
![image](/uploads/d56a668b09e90dfd7e833f1e2a09e82f/image.png)
* Goto site edit page and select legal entities.
![image](/uploads/2ee83f11890aa5c15cd806c60fea0c8f/image.png)
* Select account invoice media type **Fax** which you want to send the fax from the account edit page.
![image](/uploads/298857578338857715d68000cc26080a/image.png)
* Goto billing cycle page and click on send fax button.

Thanks.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Apr 18, 2019, 09:09

@amishra 

Thank you, I was able to setup and test this, however, I did not receive any of the faxes. I believe this is due to we need to authorize the 'From' email address to allow for it to send faxes out. What is the email address the system sends these faxes from? I would think that it is the site email address from my testing below.

**Testing:** 
* I was able to update the site to use the Demo legal entity; and I also updated the site email address to "kyle.johnson@answernet.com" as this is an authorized email to send faxes from. 
* I set the outbound fax to "answernet.com". 
* Then I set the fax number for account 7654 to "kyle.johnson". 
* I expected it to send an email then as it would go to "kyle.johnson@answernet.com" via the fax option. 
* This occurred just as expected and it came from "kyle.johnson@answernet.com". 
* I then updated the fax to "4065340399" and the legal entity to "billings.fax.answernet.com". 
* I then reprocessed the fax and sent again, it did not go through. I also did not see it in our viewmyfax portal or vitality. I am reaching out to @tim.traylor as it looks like something might be blocking it from our side. 

**Concerns:**
*  The fax field in the account edit page allows for characters and essentially non fax/phone numbers. Should we update this to validate only 10 digit numeric values?
* Found out we need to have a "1" in front of the ten digit phone number. Should we add this as part of the validation, or have this occur via programming on the back end when sending?
 * I was able to manually send via my email to the fax number successfully, but was not able to get SAB to send out still. Tim believes this is due to a DNS issue.

**Comment by kdjstudios** [comment]
In GitLab by @amishra on Apr 23, 2019, 22:54

@kdjstudios @tim.traylor 

We are still waiting for your feedback. Please test it so that we can use for other testing branches.

Thanks.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Apr 24, 2019, 09:28

@amishra @tim.traylor 

Thanks for the update. We are still waiting for our internal tech team to setup the correct domains so the DNS issue is resolved. We will keep you posted once they have completed that and we have been able to retest.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Apr 29, 2019, 08:17

changed the description

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Apr 29, 2019, 08:21

@tim.traylor wrote in the ticket:

> We need two new public subdomains defined: answernet.fax.sahosted.com and hosted.fax.answernet.com

However, will those both not need to be on sabilling.com domains? or should they both be on sahosted.com domains? I feel like having one and the other is not correct for some reason.

**Comment by kdjstudios** [comment]
In GitLab by @tim.traylor on Apr 29, 2019, 11:33

@kdjstudios Hi Kyle - you're correct and I wrote the wrong thing.  They should both be sahosted.com.  Thx!!

**Comment by kdjstudios** [comment]
In GitLab by @amishra on Apr 30, 2019, 07:24

@kdjstudios 

Do we need to use this **sahosted.com** in the code?

Thanks

**Comment by kdjstudios** [comment]
In GitLab by @tim.traylor on Apr 30, 2019, 07:50

@amishra Asish, these host names will be in the new fields, so you don't need to change anything.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Apr 30, 2019, 16:24

@amishra @pchaudhary @smasih @tim.traylor 

There are a few items we do need to update in regards to the SAB Faxing below. Please provide estimates for these updates below.

- Validation on Fax Numbers
 - Validate on account save page (NO formatting, only a 11 digit number)
- When sending the faxes out, they need to be sent from "support@sabilling.com" email address.
 - This is due to the efax service only allowing sending via this email address to prevent fraud.
 - This will also insure if we have any failed faxes via the efax service email notification will come into the support team and we can forward them accordingly.
- Optional: we do not need to have a body or subject line, just the attachment when sending out. If this is a quick and simple adjustment please do it, if it will take over 30 minutes do not do it.

**Comment by kdjstudios** [comment]
In GitLab by @amishra on Apr 30, 2019, 23:58

@kdjstudios 

Thank you for the update.

Estimate on validating fax number from both side frontend and backend.:

|Size|Lower Time Bound|Variability|Upper Time Bound|Testing|Time for Testing| Time for Dev & Testing|
|-|-|-|-|-|-|-|
|R-Small|2|Concepts and Implementation Well Understood|2.5|Low Impact|0.5|3|





> Optional: we do not need to have a body or subject line, just the attachment when sending out. If this is a quick and simple adjustment please do it, if it will take over 30 minutes do not do it.

It's a quick change and will do it.

thanks.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on May 2, 2019, 12:27

I am going to approve this since we need it.

**Comment by kdjstudios** [comment]
In GitLab by @tim.traylor on May 3, 2019, 06:02

@amishra @kdjstudios Asish, before we take any here can you pls tell me what account is being used for the from address right now?

Kyle, are the the body and subject line showing up in the received fax?  If not, let's not worry about it.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on May 3, 2019, 08:18

@tim.traylor 

The body and subject line don't show up in the fax, unless there is no attachment, then the body is sent as the fax. Which is why I would rather not risk sending the body and just have it be blank. As requested in http://gitlab.aavaz.biz/AnswerNet/SABilling/issues/260#note_39367 and @amishra replied in http://gitlab.aavaz.biz/AnswerNet/SABilling/issues/260#note_39373 I only said we should do this if it takes less then 30 minutes. 

From my previous testing for the email and the eFax it appears we are using the site's email address. I requested in http://gitlab.aavaz.biz/AnswerNet/SABilling/issues/260#note_39367 that we need to use the support@sabilling.com email, as the email to eFax service will only send from previous approved email addresses setup in viewmyfax.com; and having to enter all the sites emails and monitor changes in those seems unreasonable.

**Comment by kdjstudios** [comment]
In GitLab by @amishra on May 7, 2019, 02:07

@tim.traylor  @kdjstudios 

Thank you for the update.


> Asish, before we take any here can you pls tell me what account is being used for the from address right now?

We were using an account email address if present otherwise using site email.

Thanks

**Comment by kdjstudios** [comment]
In GitLab by @amishra on May 7, 2019, 02:28

@kdjstudios 

We have started the development. we will inform you when ready for testing.

The new functionality will be developed like this:

**Account Form page**

* Only input an 11 digit number.

**While sending a fax** : 
* Sender: "support@sabilling.com"
* Recipient: fax_number@outbound_efax_host
* No Subject.
* No Body.
* only Attachment.
* If any errors i.e "blank fax number" than email errors will behave as:
  * Sender : **Site email** and **support@sabilling.com**
  * Recipient : **Site email** and **support@sabilling.com**
  
Thanks.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on May 7, 2019, 08:41

@amishra 

Thank you, I was able to confirm receipt of the error email for blank fax numbers at the support email as it came form the support email too: http://www.servicedesk.answernet.com/profiles/ticket/8306385

Will we need to run a conversion on existing fax numbers to make them compliant with the 11 digit format before we release this?

Also what server is this on so I may test the validation on the fax number?

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on May 8, 2019, 13:20

@amishra @pchaudhary @smasih 

I reviewed this with @tim.traylor and we will want to have this update be released this weekend. I have not received any responses to the above, but would presume that we will need to run the conversion on the existing fax numbers to ensure they pass validation. 

I am still needing to confirm the validation works, what server is this on?

**Comment by kdjstudios** [comment]
In GitLab by @amishra on May 9, 2019, 03:36

@kdjstudios 

Thank you for the update.



> I am still needing to confirm the validation works, what server is this on?


We have set up on the dev server. Please test it.


> but would presume that we will need to run the conversion on the existing fax numbers to ensure they pass validation.

* We have run the script to check how many fax number greater than 11 digits including minus sign(-).

* We have attached attachment both of the server including active and inactive. Please find below attachment.
* In attachment we have status column GT (Grater than 11 digits) and LT(Less than)  

[Internal](/uploads/8dad885362c0d9dcd44b880e5c8bc476/fax_answernet_new.csv)   

[External](/uploads/d3e5d76024ba6085d77fcb0adc34488c/fax_keener.csv)   


Thanks.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on May 9, 2019, 14:59

@amishra 

I just tried to test this on dev server and first there was no demo site anymore.What happened to it?

![image](/uploads/ded1b0415b1940ab472bd902648ea980/image.png)

Second, I was able to save a fax number of only 3 digits on an account? This is clearly not working.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on May 9, 2019, 15:02

@tim.traylor 

What do you think the best path to update the existing fax numbers would be?

- I feel we should be able to automatically remove any non numeric characters and if there is 10 digits add the 1 prefix automatically. Which from the looks will resolve a majority of the accounts. However the 7 digit numbers I feel we will need operations to update manually.

**Comment by kdjstudios** [comment]
In GitLab by @amishra on May 9, 2019, 23:49

@kdjstudios 

Thank you for the update.

> I just tried to test this on dev server and first there was no demo site anymore.What happened to it?

We looked into this and I am able to see the demo site.

Please check LDAP access.




> Second, I was able to save a fax number of only 3 digits on an account? This is clearly not working.

We have fixed that and allow only 11 digit numbers.



 

> What do you think the best path to update the existing fax numbers would be?
>
> - I feel we should be able to automatically remove any nonnumeric characters and if there are 10 digits add the 1 prefix automatically. Which from the looks will resolve a majority of the accounts? However the 7 digit numbers I feel we will need operations to update manually.



* We have run the script again removing the non-numeric character. Please find below attachment.

* In attachment we have length column so that you can sort by length to see what is the actual length of fax number when removing the non-numeric character.

[Internal](/uploads/a14c83864ede550f10d90382e9c11038/fax_answernet_new_with_length.csv)  

[External](/uploads/0ae9bb5b2d0b0c1fc66233805ac33634/fax_keener_new_with_length.csv)

Please test it on the dev server.

Thanks.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on May 10, 2019, 08:44

@amishra @pchaudhary @smasih @tim.traylor 

Thank you for the update. I am not a LDAP user, but a local one. I checked in SAB and someone had removed my access from that site. I changed it back. Makes me wonder who would have done that... However, it still is not working. I can not access the demo site, even though my user says I have access to it. NOTE: I logged out and I logged back in and still not working. What is going on? Please look into this and if needed create a separate ticket. 

![image](/uploads/24a1efb9538fd925f0863d1d59f04792/image.png)

![image](/uploads/c3eae1ef357b8cc7b607d9eb565e014c/image.png)



Thank you, That will bring us down to 23 active accounts on internal that operations will need to manually check and 2 accounts on external. (Even though none of them are setup for fax deliver).

[fax_keener_new_with_length_Filtered.xlsx](/uploads/2b72d3c91c1967561437b27ecceb7491/fax_keener_new_with_length_Filtered.xlsx)

[fax_answernet_new_with_length_Filtered.xlsx](/uploads/083f7b353b607aca8c587e0af142753a/fax_answernet_new_with_length_Filtered.xlsx)


I was able to test and confirm that the fax number now will only accept 11 digit numeric input. 

@tim.traylor may you confirm below for the release. 

We will need to have the update released and the fax numbers updated to remove all non numeric characters and for 10 digit characters to have a "1" prefixed to them. I will send out an email to the sites and clients instructing them to update the remaining accounts on the lists I have attached.

**Comment by kdjstudios** [comment]
In GitLab by @pchaudhary on May 11, 2019, 04:36

@kdjstudios 

We're not releasing this ticket with today's milestone as there are some points which need still a clarification prior to release.

We'll keep this update on our Monday morning.  

Thanks

**Comment by kdjstudios** [comment]
In GitLab by @pchaudhary on May 12, 2019, 23:28

@kdjstudios @tim.traylor 

We find that before releasing this to the production servers, we need to fix all of the faulty fax numbers because this can cause discrepancy if any of the accounts validation fails at the time of accounts update.  

So, there are 2 options to go forward: 
1.  We have to fix all of the faulty fax numbers of account's (*active as well as terminated*) because of the validation.  
1.  We can update the validation code to only validate the active accounts.

Although, we prefer the later one point, please provide your inputs to move further.

Thanks

**Comment by kdjstudios** [comment]
In GitLab by @tim.traylor on May 13, 2019, 05:53

@pchaudhary Pls change the validation to active accounts only.

**Comment by kdjstudios** [comment]
In GitLab by @amishra on May 17, 2019, 01:51

@kdjstudios @tim.traylor 


Thank you for the update.

We have removed validation from inactive accounts. Please test it on the dev server.

Thanks.

**Comment by kdjstudios** [comment]
In GitLab by @amishra on May 23, 2019, 02:31

changed the description

**Comment by kdjstudios** [comment]
In GitLab by @amishra on May 23, 2019, 02:31

changed the description

**Comment by kdjstudios** [comment]
In GitLab by @amishra on May 23, 2019, 02:32

changed the description

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jun 4, 2019, 09:41

This was reviewed with Gary, about 4 hours was estimated on this. He wants to see how many accounts are using the faxing and if it is worth fixing. 

Gays wants me to contact Kravitz on how we are billing externals for their faxes.

**Comment by kdjstudios** [comment]
In GitLab by @amishra on Jun 4, 2019, 23:50

@kdjstudios 

Thank you for the update.

### Answernet

Lists of all accounts both active and inactive having invoice media with FAX:- `167`  
[answernet_account_with_inv_media_fax](/uploads/4740c7945379ed4ffbc361d694453532/answernet_new_account_with_inv_media_fax.csv)

### Hosted

Lists of all accounts both active and inactive having invoice media with FAX:- `40`  

[keener_account_with_inv_media_fax](/uploads/582cf2eaef30ac2c4a4510eeed1862b4/keener_new_account_with_inv_media_fax.csv)  


Thanks.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jun 5, 2019, 08:20

@amishra @tim.traylor 

Thank you for the update, since terminated accounts we generally do not care about, that is really only 52 internally and 3 externally. (Makes sense why most of them have not complained as they probably have not realized they do not even receive their invoice, since the faxing is not working.)

I will update with Kravitz's response to how we bill for this service once we receive it.

Marking this back for @tim.traylor review. I personally would think since we have already coded the functionality we should keep/release it in case future clients use it or we internally want to keep using it, we really are just waiting on the AnswerNet Tech guys to create the email to fax scripts/function.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jun 6, 2019, 08:42

Kravitz wrote:

> As far as I know the fax invoice option in SAB is included in the monthly recurring fee we charge them, I also know that most of our SAB clients now use email delivery for monthly invoices. Most have begun charging a fee for paper invoices mailed out. 
> 
> As for our internal clients, I believe we only charge a fee for paper invoices. I would venture to say that the sites majority of invoices go via email with a small percentage going by U.S. mail and even fewer being faxed out. 
> 
> I have added Cori onto the email to see if she concurs with me. 
> Hope this helps. 


Cori wrote:

> If you are talking strictly about faxing out of SAB (and not VCC)- we do not charge our clients for receiving invoices via Fax. The only fee we have is for Paper Invoicing. Cori  

In summary, we are not billing for any fax usage out of SA Billing. However, for our SA Billing clients we include that service as part of their base rate.

**Comment by kdjstudios** [comment]
In GitLab by @amishra on Jun 21, 2019, 24:41

@kdjstudios 

Thank you for the update.

The development & testing is completed in this issue.
We have run the script for validating/checking on fax number and there are still some accounts which need to update on both of the servers. 

Please find below attachment. 

We will release this once all of the accounts fixed.(Both internal & external server)

### ANSWERNET (22 accounts.)

[answernet.csv](/uploads/9161cdd1847fcc8f07d187ead1e5db44/answernet.csv)
### Hosted (2 accounts.)

[keener.csv](/uploads/1b6d3a39778c77b10169f4088e368dd4/keener.csv)


Thanks.

**Comment by kdjstudios** [comment]
In GitLab by @pchaudhary on Jul 5, 2019, 01:03

removed milestone

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Sep 27, 2019, 11:49

Gary is reaching out others to help figure out what the hold up is on getting this fixed.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Dec 12, 2019, 11:30

@amishra 

I just want to confirm that we still are not able to send faxes out of SA Billing currently?

If I reviewed the ticket correctly, we have developed an update to allow for using the efax, but that has not been published yet correct?

**Comment by kdjstudios** [comment]
In GitLab by @amishra on Dec 15, 2019, 22:43

> I just want to confirm that we still are not able to send faxes out of SA Billing currently?

*  Yes, Currently SAB does not support.

> If I reviewed the ticket correctly, we have developed an update to allow for using the efax, but that has not been published yet correct?

* Yes, We have developed the functionality and waiting for the update of the fax number from the account in my last comment.


**Do we need to run a script to check how many accounts have still invalid fax number?**

Thanks

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Dec 16, 2019, 11:42

@amishra 

Thank you for the update. may we please have this pushed back to testing. Our networking team believes they are close to having the eFax accounts/email setup. 

Where/when did we discover there was/is invalid fax numbers?

**Comment by kdjstudios** [comment]
In GitLab by @amishra on Dec 17, 2019, 24:11

@kdjstudios @jredfield 

Thank you for the update.

> Thank you for the update. may we please have this pushed back to testing. Our networking team believes they are close to having the eFax accounts/email setup.

* Before pushing back for your testing, we need to additional **2-3 hrs** to update with the latest code/branch since it's almost 5 months old.


> Where/when did we discover there was/is invalid fax numbers?

Please see my on [comment](http://gitlab.aavaz.biz/AnswerNet/SABilling/issues/260#note_42689)

Thanks

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Dec 17, 2019, 10:44

@jredfield 

2-3 hours to test with updated code seems extremely high, I would think this would only take 1 hour tops. Unless that includes that time to also run the fix on the bad fax numbers.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Dec 19, 2019, 12:23

We need this functionality so moving to development. but this should easily be accomplished in under the 2 hour estimate.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Dec 31, 2019, 14:10

Updating the priority. Our techs were able to work out the efaxing accounts and we now should be able to deliver efaxes from SA Billing. 

However, I did notice that the efax accounts restrict which emails the faxes can come from. So I would like to confirm that when we send out efaxes from SAB they need to come from "Support@sabilling.com" on both servers.

**Comment by kdjstudios** [comment]
In GitLab by @ranand on Jan 2, 2020, 02:31

@kdjstudios @jredfield 

There are still some invalid fax numbers on both the servers. Please see the attached CSVs 

[invalid_fax_number_answernet.csv](/uploads/1780399cc70d1bfb289d7a75ab79e822/invalid_fax_number_answernet.csv) **(14 accounts)**

[invalid_fax_number_hosted.csv](/uploads/b57c58f03d1cce9b12061b4165e8f6ff/invalid_fax_number_hosted.csv) **(2 accounts)**

Thanks

**Comment by kdjstudios** [comment]
In GitLab by @amishra on Jan 2, 2020, 05:45

@kdjstudios @jredfield 

Thank you for the update.

We will set up branch for your testing tomorrow.

Thanks

**Comment by kdjstudios** [comment]
In GitLab by @amishra on Jan 3, 2020, 01:10

@kdjstudios @jredfield 

We have setup branch on the merge2 server for your testing . 

* We have tested using fax number (**4065340399** or **14065340399**) with outbound_efax_host as **billings.fax.answernet.com**. may I know did you get any mail/fax?  

* Test cases given above on my [comment](http://gitlab.aavaz.biz/AnswerNet/SABilling/issues/260#note_38611).
 
**Account Form page**
* Only input an 11 digit number.

**While sending a fax** : 
* Sender: "support@sabilling.com"
* Recipient: fax_number@outbound_efax_host
* No Subject.
* No Body.
* only Attachment.
* If any errors i.e "blank fax number" than email errors will behave as:
  * Sender : **Site email** and **support@sabilling.com**
  * Recipient : **Site email** and **support@sabilling.com**

Thanks

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jan 3, 2020, 10:51

@amishra @jredfield 

Where is the"outbound_efax_host" specified? I didn't see it in any settings in the UI, so I presume we are hard coding it, which is fine, I just need to know which host this server is setup to use to confirm it is working with both.
- We should be using "answernet.fax.sahosted.com" for the internal server and "hosted.fax.sahosted.com" for the external server. 

During sending if there is a existing "bad" fax number then it should appear as an error on the Send Fax email notification. As shown below, it would appear we were successfully able to send to a fax number with dashes, which I believe is not invalid? We should have an Error of "Invalid Fax Number Format".

![image](/uploads/ce345198a2bdca27b1ca1f30ab03629e/image.png)

This confirmation pop up should state "Please confirm to send out Fax Invoices."

![image](/uploads/0d8c4c88de17303993accfcdb2bd99d3/image.png)


I setup the above 4 accounts with fax numbers, and clicked the send fax button, but did not receive any of the faxes. I also did not see any faxes go out. Which I believe may be caused to us not having the correct hosts setup. Please confirm. 

![image](/uploads/37330b4d1557fa8eda220cdf78f0b245/image.png)

**Comment by kdjstudios** [comment]
In GitLab by @amishra on Jan 3, 2020, 11:09

@kdjstudios @jredfield 

> Where is the"outbound_efax_host" specified? I didn't see it in any settings in the UI, so I presume we are hard coding it, which is fine, I just need to know which host this server is setup to use to confirm it is working with both.

* Please look at the legal entity page under the admin section.

![image](/uploads/390b79aaa55528d8e7fb147b2fafbf8f/image.png)

* We have mentioned the above test cases on my [comment](http://gitlab.aavaz.biz/AnswerNet/SABilling/issues/260#note_38611). Please have a look.

Thanks

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jan 3, 2020, 12:02

@amishra 

Thank you for the update. I was able to setup the demo site to use the Demo Legal Entity and have my email for the site email. I then removed all the fax accounts except for the Demo 54 and T-DEMO-TEST. I then resent using both of the above Hosts. No invoices were sent out and I do not see them listing in the efax account. So it would appear that they are not making it out of SA Billing; However, I did not receive any error emails either.  

I confirmed the hosts/accounts are working by sending a email (From my sahosted email account to insure it was sending from the sahosted domain - kyle.johnson@sahosted.com) to both of the fax accounts. Those sent correctly so it would appear there is no issue with the efax accounts. I also want to confirm we are using the smtp.sahosted.com to send these email/faxes.

Please use the following for testing. 

* 16025815206@hosted.fax.sahosted.com To test the external fax, by sending a fax back to itself. 
* 16025815205@answernet.fax.sahosted.com to test the internal fax, by sending a fax back to itself.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jan 3, 2020, 12:06

I did eventually receive these error emails. I am reaching out to our tech guys also to see if they can determine what it means. 

![image](/uploads/398f5fd5b00c52fd469ea7394e443af0/image.png)

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jan 3, 2020, 14:07

May we please add our tech to the CC option on these too, both emails listed below. That way he can look at the raw mime of the email to determine why it is not sending. 

ben.carrasco@answernet.com and ben.carrasco@gmail.com

**Comment by kdjstudios** [comment]
In GitLab by @amishra on Jan 4, 2020, 03:53

@kdjstudios @jredfield 

Thank you for the update.

We have added both email addresses while faxing and also setup a branch for the testing so that you can test as you want.

Thanks

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jan 6, 2020, 10:30

@abakshi 

I got the following error when trying to click the "Send Fax" button.

![image](/uploads/0bc30a56b4e0676e403875670d5b82a3/image.png)

About 20 minutes later I did receive this email, which is a bit puzzling since I encountered the above error. NOTE: I just saw the timestamp on the email is at 5:48am, but I didnt get it until 11:49am. Is that because it's using the server's timezone or something? Or was this a delayed email from your testing earlier today?

![image](/uploads/f1c02e38cd9e8a25e4e512401ee958b9/image.png)

**Comment by kdjstudios** [comment]
In GitLab by @amishra on Jan 6, 2020, 11:12

@kdjstudios @jredfield 

Thank you for the update.



> I got the following error when trying to click the "Send Fax" button.

Yes, this was the syntax issue but while our testing we did not get it.
Although, we have fixed the issue Please test it again on the same server.

thanks

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jan 6, 2020, 12:18

Thanks, Our tech received the emails CC and is looking into why they are not going out on the fax currently. I will update with his response accordingly.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Feb 18, 2020, 15:39

I followed up with Ben for an update today.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jun 5, 2020, 11:47

@amishra @ranand @ranand 

We are still waiting for updates from Answernet's infrastructure team to ensure everything is working on our side.

May we please confirm that everything is working on the SAB side of things?

**Comment by kdjstudios** [comment]
In GitLab by @amishra on Jun 11, 2020, 24:35

@kdjstudios @Robleski 

Thank you for the update.

> We are still waiting for updates from Answernet's infrastructure team to ensure everything is working on our side.

* Please let us know and then we will setup branch for your testing.



> May we please confirm that everything is working on the SAB side of things?

* We need to merge with the latest code because this branch is approx 3 months old. 
* We need to 2-3 hrs for updating with latest code and test too.

## Developer Estimate
|Size|Lower Time Bound|Variability|Upper Time Bound|Testing|Time for Testing| Time for Dev & Testing|
|-|-|-|-|-|-|-|
|R-Small|2|Concepts and Implementation Well Understood|2.5|Low Impact|0.5|3|

##### Estimate Notes:

* Merge with the latest code.
* Solve conflict if any
* testing.



Thanks

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jun 11, 2020, 09:15

Please push this to a Dev server that we can have this branch on for at least a week to allow for time for the Answernet Infrastructure team to test.

**Comment by kdjstudios** [comment]
In GitLab by @amishra on Jun 12, 2020, 01:36

@kdjstudios @Robleski 

Thank you for the update.

We have updated the latest code and setup branch on the dev server for your testing. Please test it and let me know.

Thanks

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jun 15, 2020, 10:22

@amishra @Robleski @rranjan @ranand 

Thank you, Please do not make any changes then to this server/branch until we have feedback from out infrastructure team confirming it is working or not working.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jun 22, 2020, 09:12

changed title from **SAB and eFax{- by Vitality integration-}** to **SAB and eFax**

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jun 22, 2020, 09:25

@amishra @ranand @Robleski 

I just received an email from Ben on our infrastructure team that the efax and gateways have been setup and are working. So I went to test this and when I tried to do the no usage step on the billing cycle I received the following error. Please fix this so we can test.

![image](/uploads/b64d644f91c7c3adcc7fb6a07747585d/image.png)

EDIT: I was able to delete the billing cycle and then update the accounts on the previous billing cycle to use the two fax numbers for each account. I then did the Sedn Faxes and we received errors for each account that the fax was not sent. I sent the following to Ben. Which appear to me that something is missing. So please double check that we are actually attaching the PDF invoices to the email/fax that SAB is sending out.

[Fax_failure_to_16025815206_from_site_number_1.msg](/uploads/9f0f38d4621a4f2fab2d1138010bae29/Fax_failure_to_16025815206_from_site_number_1.msg)

[Fax_failure_to_16025815205_from_site_number_1.msg](/uploads/d8590d18c372640b9506dcdf92490023/Fax_failure_to_16025815205_from_site_number_1.msg)

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jun 22, 2020, 10:18

Also may we please have a copy of one of the the email/faxes that SAB is sending out. I know we have the restrictions on the fax number field to only allow for a 11 digit fax number, so I can't; but may someone temporarily bypass that so we can receive it via an email account. I think that may help Ben confirm what the gateway is receiving and further diagnose the issues. If so, please send an email "Fax" to: kyle.johnson@answernet.com and ben.carrasco@answernet.com. Please send at least one email "Fax" from each server/gateway (http://gitlab.aavaz.biz/AnswerNet/SABilling/issues/260#note_53883).

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jun 22, 2020, 11:17

@amishra @ranand @rranjan @Robleski 

Please ignore the above. Ben was able to get it working and found it was an issue with the file name and MIME headers. Which is over my head on understanding. But we were able to successfully send out faxes and they were sent from the SABin (Internal) account. Ben is also updating the SABex (External) account to use the same updated processing. So if we can please have this now setup for testing the External account/domain. would be wonderful.

**Comment by kdjstudios** [comment]
In GitLab by @rranjan on Jun 23, 2020, 07:30

@kdjstudios  @Robleski 

Thanks for the update.

We have set up a branch on the merge server for your testing.


Thanks

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jun 23, 2020, 13:08

@rranjan @amishra @ranand @Robleski 

This does not appear to be working with the other gateway. It looks as though the other gateway is not setup in SAB? How did your team confirm this was working?

![image](/uploads/1bec5c0ca6ad506e56bac100cfdc677e/image.png)

**Comment by kdjstudios** [comment]
In GitLab by @amishra on Jun 24, 2020, 07:11

@kdjstudios @Robleski 

Thank you for the update.

* We would like to inform you did not fill the **Outbound Efax Host** from the Admin section.

Although, We have added **** in the Outbound Efax Host under the legel entity. you can change accordingly.


![image](/uploads/213e95f6c28bc60ee9588853ee59febc/image.png)


Thanks

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jun 24, 2020, 09:14

@amishra @Robleski 

How were you able to test this if that was not filled in? Seems to me your team did not test this and confirm it was working before sending to me for testing, otherwise your team would have found that issue that would have been filled in. That is exactly what we requested for your team to do in  http://gitlab.aavaz.biz/AnswerNet/SABilling/issues/260#note_60214. 

I was able to test this with Valid/Real fax numbers and confirm it is working now. Marking for ready. 

When releasing this to production please make sure all sites/legal entities on the internal server get assigned "answernet.fax.sahosted.com" while all sites/legal entities on the external server get assigned "hosted.fax.sahosted.com" as their outbound efax host.

**Comment by kdjstudios** [comment]
In GitLab by @amishra on Jun 26, 2020, 06:47

@kdjstudios @Robleski 

Thank you for the update.

We have re-run the script again and found Answernet server has still 11 accounts having an invalid fax number and hosted has 2 accounts. Please find the below attachment.

**NOTE: All accounts have different invoice media.**

##### Answernet 

[answernet_fx_lists.csv](/uploads/e73bf153f19ed4eaf6904fed4b40ee6d/answernet_fx_lists.csv)

##### Hosted

[hosted_server_fax_lists.csv](/uploads/fc2ef8afaec4a1d72ca155db28b8ffd0/hosted_server_fax_lists.csv)


* So we two options to fix it.
  1. If we will release this ticket tomorrow, then we need to remove the fax number from all the accounts.
  2. we will wait for release until all accounts are not fixed.

Please confirm 

Thanks

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jun 26, 2020, 10:20

@amishra @Robleski @ranand @rranjan 

Please release this with out fixing the fax numbers. As it should be fine to leave those as is. As part of this release SAB will send out an error email to the site to notify them when SA billing tries to send a fax to those invalid numbers to allow them to fix them. Also if the ops/client tried to edit the account they will be presented with an error to fix them. 

NOTE: Please keep in mind faxing has not been working in SA Billing for 2-3 years now, so most likely the site/client was manually sending those faxes to their clients, and may continue to manually send them.

**Comment by kdjstudios** [comment]
In GitLab by @amishra on Jun 27, 2020, 04:11

@kdjstudios @Robleski 

Thank you for the update.

We have successfully released on both of the production servers.

`NOTE:`
 * We have removed backend validation for the fax number.once fix all the account we will add validation on backend level too.**

 * We have run the script for active account only.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jun 27, 2020, 08:25

I was able to confirm in the admin section that the legal entities now all have an efax host set. I also did one test sending of efaxes on the demo accounts.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jun 30, 2020, 10:17

closed
